WordPress/gutenberg
# Make child themes inherit parent's style variations.

## What?


Makes parent theme style variations inherited by the child theme.

cc @WordPress/block-themers 

## Why?


Closes #45965

## How?


Changes the `get_style_variations` method of the theme json resolver. Checks if it's a child theme and returns the parent theme variations in addition to the child's. If the file basename of the child variation is the same as its parent, it will only include the child's ("overwriting" the parent variation).

## Testing Instructions





1. Activate a child of a theme that has style variations. You can use this TT3 child for testing: https://github.com/WordPress/theme-experiments/pull/311
2. Verify the child theme's variations appear in addition to the parent themes:

Before | After
--- | ---
<img width="1624" alt="Screenshot 2022-12-16 at 12 41 06 PM" src="https://user-images.githubusercontent.com/5375500/208157099-645202aa-fef5-4d87-ba52-f0060b7aaa92.png"> | <img width="1624" alt="Screenshot 2022-12-16 at 12 38 26 PM" src="https://user-images.githubusercontent.com/5375500/208157146-937db5df-a6e7-4e93-b2b7-0dfd1fe489dd.png">


### Testing Instructions for Keyboard


## Screenshots or screencast
